planner/core/rule_join_elimination.go

```diff
@@ -51,12 +51,20 @@ func (o *outerJoinEliminator) tryToEliminateOuterJoin(p *LogicalJoin, aggCols []
 	for _, outerCol := range outerPlan.Schema().Columns {
 		outerUniqueIDs.Insert(outerCol.UniqueID)
 	}
-	matched := IsColsAllFromOuterTable(parentCols, outerUniqueIDs)
-	if !matched {
-		return p, false, nil
+
```

Comment:
Please update function comment together.
"For example left outer join, if the parent don't use any right table columns and xxxx"

Reply:
done